### ISSUE TITLE
Saving templates - remind users to save

### ISSUE THREAD

**Issue by dsisu** [DMPRoadmap/roadmap#2906]
Please complete the following fields as applicable:

**What version of the DMPRoadmap code are you running? (e.g. v2.2.0)**
3.0.2

**Expected behaviour:**
Admin users should be reminded to save changes to templates. Some of them forget to do it and there is no warning mechanism in place to point this out.

**Actual behaviour:**
Users can move away from template (close window) without warning of unsaved changes.
Users can add question to a section without saving, then add new section to the template and save. It saves the new section and doesn't warn about unsaved changes (questions) to previous section.

**Steps to reproduce:**
Edit an existing test template to add questions and sections and test saving procedure.

**Comment by dsisu** [comment]
There is a wider issue with varying saving procedures in DMP Roadmap. Saving answers to questions in plans leaves the plan open and fully expanded, even when answering questions from different sections. Saving changes to questions in a section of the template, collapses previous sections (so users are unaware of losing any changes they made to previous section).

**Comment by dsisu** [comment]
Solutions proposed by users were:
- dynamic saving
- a dashboard for administrative users, which lists all functionalities available to them, and highlights unpublished work, although note that published-unpublished may have a different meaning when it comes to templates, for example, where unpublished means 'not made public' rather than 'has unsaved changes'.

**Comment by dsisu** [comment]
This issue arose out of 2020 Usability Testing exercise.

**Comment by briri** [comment]
Agreed that we need a standardized way of saving across the site. The only place we auto-save is on the Plan.

**Comment by dsisu** [comment]
Other issues regarding saving procedure, which also arose out of the usability testing:
- while editing a plan, if you answer a question then hit save, the question remains open and the screen remains fixed on the question
- while editing a template, if you edit a question and hit save, it closes and the screen jumps to the top of the phase details
- while editing guidance, if you edit a guidance item then hit save, guidance remains open but screen jumps to the top of the page.

The behaviour should be the same, for example, sections, questions always collapse OR implement an editing vs viewing screen, to make it obvious to the user whether changes have been saved or not.

**Comment by dsisu** [comment]
Users also commented that perhaps it would be useful to have a Save All button (for when user edits several items on the page). Apparently, the Save button acts like a Save All but this is not obvious to the user.
